File: 2020/02/javascript/part1.js
let fs = require('fs');
const input = fs.readFileSync('./input.txt', 'utf-8').toString();

inputArray = input.split('\n').map((value, index) => {
  return value.split(' ');
});
console.log(inputArray.length);
let goodPassWordsCount = 0;

for (let passsword of inputArray) {
  console.log(passsword);
  let passwordLength = passsword[2]?.length;
  console.log('PASSWORD', passsword[2]);
  let pw = passsword[2];
  let counter = 0;
  for (let char = 0; char <= passwordLength; char++) {
    let character = pw[char];

    if (character == passsword[1][0]) {
      counter++;
    }
  }
  if (
    counter >= passsword[0].split('-')[0] &&
    counter <= passsword[0].split('-')[1]
  ) {
    goodPassWordsCount += 1;
  }
}

console.log(goodPassWordsCount);
console.log(inputArray.length);

File: 2020/02/python/part1.py
with open("../input.txt") as file:
    input = file.read()
file.close()

passwords_list = input.split("\n")
passwords_list.pop()

passwords_list = [pw.split(" ") for pw in passwords_list]

good_pw_count = 0;

for password in passwords_list:
    pw_length = len(password[2])
    pw = password[2]
    pw_count = pw.count(password[1][0])

    if pw_count  >= int(password[0].split('-')[0]) and pw_count <= int(password[0].split('-')[1]):
        good_pw_count += 1

print(good_pw_count)
